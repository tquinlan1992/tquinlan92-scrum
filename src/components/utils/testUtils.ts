import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { AnyAction, isType, ActionCreator } from 'typescript-fsa';
import { AppState } from '../../headless/store';

export function getAnyJestFn() {
    return jest.fn() as any;
}

interface AnyActionPayload extends AnyAction {
    payload: any;
}

export function expectActionWithPayload(actionToTest: AnyActionPayload, expectedActionCreator: ActionCreator<any>, expectedPayload?: any) {
    expect(isType(actionToTest, expectedActionCreator)).toBe(true);
    expect(actionToTest.payload).toEqual(expectedPayload);
}


const middlewares = [thunk];
export const mockStore = configureStore(middlewares);

export function getMockStore(state: Partial<AppState>) {
    return mockStore(state);
}

export function expectCalledOnceWith(mockFunction: jest.Mock<{}>, calledWith?: any) {
    expect(mockFunction).toHaveBeenCalledTimes(1);
    const expectedParams = calledWith ? [calledWith] : [];
    expect(mockFunction.mock.calls[0]).toEqual(expectedParams);
}

export function mockClearAll(mockFunctions: jest.Mock<{}>[]) {
    mockFunctions.forEach(mockFunction => {
        mockFunction.mockClear();
    })
}