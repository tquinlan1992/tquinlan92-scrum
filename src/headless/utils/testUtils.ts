export function mockPouchDB() {
    jest.mock('../database/core/pouch', () => {
        return {};
    });
}
