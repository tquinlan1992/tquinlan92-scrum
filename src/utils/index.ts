import * as _ from 'lodash';
import { connect, MapStateToPropsParam, MergeProps } from 'react-redux';
import { AppState } from '@headless/store';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';

type PropsMap<M, K extends keyof M> = {[P in K]: M[P]};

export function pick<M extends object, K extends keyof M>(obj: M, ...props: K[]): PropsMap<M, K> {
    // @ts-ignore: TS2352: Type 'PartialDeep<M>' cannot be converted to type 'PropsMap<M, K>'.
    return _.pick<M>(obj, props) as PropsMap<M, K>;
}

export type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T];

export type AppStateThunkDispatch = ThunkDispatch<AppState, undefined, AnyAction>;
