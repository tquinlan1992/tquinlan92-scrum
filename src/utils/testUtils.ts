export function mockPouchDB() {
    jest.mock('../headless/database/pouch', () => {
        return {};
    });
}

export function mockPouch(dbMethods: object) {
    jest.mock('../headless/database/pouch', () => {
        return {

        }
    })
    const mockPouchRequire = require('../headless/database/pouch');
    const mockGetRemoteDb = jest.fn(() => {
        return new Promise(resolve => {
            resolve(dbMethods);
        })
    });
    mockPouchRequire.getRemoteDb = mockGetRemoteDb;
    return mockPouchRequire;
}

export function mockShortId() {
    jest.mock('shortid', () => {
        let count = 0;
        return () => {
            count = count + 1;
            return String(count);
        }
    })
}

