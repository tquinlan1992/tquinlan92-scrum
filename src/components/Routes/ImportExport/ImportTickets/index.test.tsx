import { ImportTickets } from './';
import { shallow } from 'enzyme';
import * as React from 'react';
import { getAnyJestFn } from '../../../../utils/testUtils';

describe('test snapshot of component', () => {
    it('should render', () => {
        const props = {
            importTickets: getAnyJestFn()
        };
        const result = shallow(<ImportTickets {...props} />);
        expect(result).toMatchSnapshot();
    });
});
