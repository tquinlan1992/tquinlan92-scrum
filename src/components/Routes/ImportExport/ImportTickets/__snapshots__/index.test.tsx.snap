// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot of component should render 1`] = `
<Fragment>
  <input
    hidden={true}
    id="file-upload-button"
    onChange={[Function]}
    type="file"
  />
  <label
    htmlFor="file-upload-button"
  >
    <WithStyles(Button)
      component="span"
    >
      Import
    </WithStyles(Button)>
  </label>
</Fragment>
`;
